### ISSUE TITLE
defined behaves differently in different implementations

### ISSUE THREAD

**Issue by djmitche** [taskcluster/json-e#342]
The following test case:
```yaml
---
title: 'defined with empty variable from a let expression'
context: {}
template: {$let: {var: 10}, in: {$if: "defined('var')", then: 't', else: 'f'}}
result: t
```
fails on JS and Python, but succeeds on Golang.  The test case demonstrates the expected behavior, though: a variable defined with `$let` should be visible to `defined`. I think the issue is that in the JS and Python implementations, the `context` variable that is being referred to in the implementation of `defined` is the original, top-level `context` to which the builtins were added, rather than the `context` that's active at the time the expression is evaluated.

Since this behavior differs between implementations, it will be important to fix this before we make another release.

@Nicrii is there any chance you could have a look?

**Comment by owlishDeveloper** [comment]
Can you tell us more about the practical problem that's behind this test case? I'm asking because one way of going about checking whether a variable is defined would be to use the concept of `truthy`/`falsy`, as in javascript language:

```javascript
if (var) return 't'; else return 'f'
```

as opposed to using `defined` built-in.... but maybe I am missing something?

**Comment by owlishDeveloper** [comment]
ok I went back down the issues and PRs, makes sense now!

**Comment by djmitche** [comment]
@Nicrii fixed this in #345.
